locations/storefinders/sweetiq.py

```diff
@@ -73,3 +76,18 @@ def parse(self, response):
 
     def parse_item(self, item, location, **kwargs):
         yield item
+
+    def storefinder_exists(response: Response) -> bool | Request:
+        # Example: https://locations.thepaperstore.com/
+        if response.xpath('//script[contains(text(), "__SLS_REDUX_STATE__")]').get():
+            return True
+
+        if response.xpath('//script[contains(@src, "sls-cdn.sweetiq.com")]').get():
+            return True
+
+        return False
+
+    def extract_spider_attributes(response: Response) -> dict | Request:
+        return {
+            "allowed_domains": [urlparse(response.url).netloc],
```

Comment:
The SweetIQ spider needs `start_urls` defined as the full URL to the location page where `response.xpath('//script[contains(text(), "__SLS_REDUX_STATE__")]').get()` is not `None`.

I think the extracted attribute then is simply just `"start_urls": [response.url]`?

Reply:
Fixed/pushed

Reply:
I think setting `allowed_domains` here to just the response domain name could break things, because in `SweetIQ.parse_locations_list` the API endpoint URL is extracted from the store locator page. I think we also need to copy `SweetIQ.parse_locations_list` in extracting the relevant API endpoint URL (domain name) here too.